client/services/FeedbackGeneratorServiceSpec.js

```diff
@@ -92,4 +99,149 @@ describe('FeedbackGeneratorService', function() {
+      expect(paragraphs[0].isTextParagraph()).toBe(true);
+      expect(paragraphs[0].getContent()).toBe(buggyOutputTestDict.messages[0]);
+
+      var feedback = FeedbackGeneratorService._getBuggyOutputTestFeedback(
+        buggyOutputTest, codeEvalResultWithSameBug).getParagraphs();
+
+      expect(paragraphs.length).toEqual(1);
+      expect(paragraphs[0].isTextParagraph()).toBe(true);
+      expect(paragraphs[0].getContent()).toBe(buggyOutputTestDict.messages[0]);
+    });
+
+    it(['should return the same hint multiple times for buggy outputs, ',
+        'provided a new error happened in between'].join(''), function() {
+      var buggyOutputTestDict = {
+        buggyFunction: 'AuxiliaryCode.countNumberOfParentheses',
+        messages: [
+          ["Try running your code on '))((' on paper. ",
+           'Did you expect that result?'].join(''),
+          [
+            'Are you making sure the parentheses are properly balanced? () ',
+            'is balanced, but )( is not.'
+          ].join(''),
+          [
+            "It looks like you're counting the number of parentheses, and if ",
+            "you have the same # of each kind, returning true. That's not ",
+            "quite correct. See if you can figure out why."
+          ].join('')
+        ]
+      };
+      var buggyOutputTest = BuggyOutputTestObjectFactory.create(
+        buggyOutputTestDict);
+      var codeEvalResult = CodeEvalResultObjectFactory.create(
+        'some code', 'same output', [], [true], [], null, 
+        null);
+      var codeEvalResultWithNewError = CodeEvalResultObjectFactory.create(
+        'other code', 'some output', [], [], [], 'ERROR MESSAGE', 'testInput');
+      var codeEvalResultWithSameBug = CodeEvalResultObjectFactory.create(
+        'new code', 'same output', [], [true], [], null, 
+        null);
+
+      var feedback = FeedbackGeneratorService._getBuggyOutputTestFeedback(
+        buggyOutputTest, codeEvalResult)
+      var paragraphs = feedback.getParagraphs();
+      TranscriptService.recordSnapshot(SnapshotObjectFactory.create(
+        codeEvalResult, feedback));
+
+      expect(paragraphs.length).toEqual(1);
+      expect(paragraphs[0].isTextParagraph()).toBe(true);
+      expect(paragraphs[0].getContent()).toBe(buggyOutputTestDict.messages[0]);
+
+      var unusedRuntimeErrorFeedback = (
+        FeedbackGeneratorService._getBuggyOutputTestFeedback(
+          buggyOutputTest, codeEvalResultWithNewError));
+      TranscriptService.recordSnapshot(SnapshotObjectFactory.create(
+        codeEvalResultWithNewError, unusedRuntimeErrorFeedback));
+
+      var feedback = FeedbackGeneratorService._getBuggyOutputTestFeedback(
+        buggyOutputTest, codeEvalResultWithSameBug).getParagraphs();
+
+      expect(paragraphs.length).toEqual(1);
+      expect(paragraphs[0].isTextParagraph()).toBe(true);
+      expect(paragraphs[0].getContent()).toBe(buggyOutputTestDict.messages[0]);
+    });
+  });
```

Comment:
I added a test for making sure that the message repeats at the end of a list, but other than that it seems like it should behave the same throughout?